### PR TITLE
Add resource, category factories and resources request specs.

### PR DESCRIPTION
**Pull request by jfhamlin**
- Adds factory_girl_rails gem.
- Adds two simple request specs to test the basic resource API.

### CODE REVIEW

**Comment by lelandg559** [comment]
has merge conflict fyi, just the gemfile stuff i think

**Comment by jfhamlin** [comment]
Ah, it must have a different version of the gemfile commit. Will fix.

**Comment by jfhamlin** [comment]
@lelandg559, updated!

File: Gemfile [hunk 1]
```diff
@@ -17,10 +17,16 @@ group :development, :test do
...
   gem 'spring'
-  gem 'rspec-rails'
 end
 
 group :development do
   gem 'rubocop', '~> 0.38.0', require: false
 end
+
+group :test do
+  gem 'factory_girl_rails', '~> 4.0'
```
Comment (lelandg559) [inline Gemfile]: im getting errors running these tests, it says 4.0 here but 4.6 in the Gemfile.lock , maybe thats the issue?

Comment (jfhamlin) [inline Gemfile]: The '~>' means that bundle can install any minor version greater than or equal to 4.0, so 4.6 is a legal choice. I'm not sure what's up with rubymine... is there a full callstack you can share? I do expect this is a problem with your dev environment though. (Travis CI and) I just run `bundle exec rake spec` to run the tests. I can successfully start a server with `bundle exec rails s`, too.

Comment (lelandg559) [inline Gemfile]: Good now, had to have rubymine bundle install manually
LGTM

**Pull request merged**